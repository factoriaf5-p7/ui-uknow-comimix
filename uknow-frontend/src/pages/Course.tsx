import Navbar from '../components/navbar/Navbar';
import { useLocation } from 'react-router-dom';
import CourseDetails from '../components/course/CourseDetails';
import Footer from '../components/footer/Footer';
import CourseHeader from '../components/course/CourseHeader';
import CourseContent from '../components/course/CourseContent';
import CommentsSection from '../components/course/CourseCommentsSection';
import CourseNewComment from '../components/course/CourseNewComment';


const Course = () => {

  const location = useLocation()
  const course = location.state;

  return (
    <>
    <div style={{padding : ' 0 0 7% 0'}}>
      <Navbar />
      <CourseHeader course={course}/>
      <CourseDetails course={course}/>
      <CourseContent course={course}/>
      <CommentsSection courseId={course._id}/>
      <CourseNewComment course={course}/>
      <Footer/>
      </div>
    </>
  );
}

export default Course;