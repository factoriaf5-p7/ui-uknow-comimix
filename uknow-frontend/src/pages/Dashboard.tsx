
import DashboardTab from '../components/elements/DashboardTab'
import Footer from '../components/footer/Footer'
import Navbar from '../components/navbar/Navbar'

function Dashboard() {
  return (
    <>
      <Navbar/>
      <DashboardTab />
      <Footer/>
    </>
  )
}

export default Dashboard