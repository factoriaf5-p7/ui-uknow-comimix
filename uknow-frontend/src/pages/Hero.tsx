import { HeroImage } from "../components/elements/HeroImage"

function Hero() {
  return (
    <>
      <HeroImage/>
    </>
  )
}

export default Hero