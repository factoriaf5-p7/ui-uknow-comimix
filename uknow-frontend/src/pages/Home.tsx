import { useContext, useState } from 'react';
import CourseList from "../components/course/CourseList";
import SearchBar from '../components/elements/SearchBar';
import uknowImg from '../assets/uknow.png';
import { Container } from "@mui/system";
import { useAllCourses } from "../services/useQuery-AllCourses";
import { CourseData } from '../interfaces/course.interface';
import Navbar from '../components/navbar/Navbar';
import Footer from '../components/footer/Footer';
import { AuthContext } from '../context/AuthContext';

function Home() {
  const { isLoggedIn, user } = useContext(AuthContext); 
  const { isLoading, isError, courseList: allCourses } = useAllCourses();
  const [courses, setCourses] = useState<CourseData[]>([]);
  const [isSearching, setIsSearching] = useState(false);
 

  // Function to filter out courses that the logged-in user has already bought or created
  const filterUserCourses = (courses: CourseData[] | undefined) => {
    if (!isLoggedIn || !user || !courses) {
      return courses; // If no user is logged in, return all courses
    }

    // Filter out courses that the user has already bought or created
    const filteredCourses = courses.filter(course => {
      const isBought = user.bought_courses.some(
        (boughtCourse: { course_id: string }) => boughtCourse.course_id === course._id
      );
      const isCreated = user.created_courses.includes(course._id);
      return !isBought && !isCreated;
    });
  
    return filteredCourses;
  };

  // Handle the search results as before
  const handleAllCourses = (searchResults: CourseData[] | undefined = []) => {
    const filteredSearchResults = filterUserCourses(searchResults);
    if (filteredSearchResults) {
      setCourses(filteredSearchResults);
      setIsSearching(true);
    } else {
      console.error("Error: filteredSearchResults is undefined.");
    }
  };
  
  

  // Filter the courses based on the logged-in user data
  const filteredCourses = filterUserCourses(allCourses);


  return (
    <div>
      <Navbar />
      <img src={uknowImg} alt="Uknow image" style={{ width: '100%', height: 'auto' }} />

      <Container sx={{ display: 'flex', justifyContent: 'center', my: 2 }}>
        <SearchBar onSearch={handleAllCourses} />
      </Container>

      {isLoading ? (
        <div>Loading...</div>
      ) : isError ? (
        <div>Error fetching data</div>
      ) : (
        <CourseList courses={isSearching ? courses : filteredCourses} />
      )}
      <Footer />
    </div>
  );
}

export default Home;