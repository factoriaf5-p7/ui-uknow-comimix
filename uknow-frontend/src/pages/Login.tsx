import Footer from "../components/footer/Footer";
import {LoginForm} from "../components/forms/LoginForm";

export default function Login() {
  return (
    <div>
      <LoginForm />
      <Footer/>
    </div>
  )
}
