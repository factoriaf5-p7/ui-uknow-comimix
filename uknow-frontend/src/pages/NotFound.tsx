import { Link } from "react-router-dom"
import Footer from "../components/footer/Footer"

function NotFound() {
  return (
    <>
      <div style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', flexDirection: 'column' }}>
        <h1>404 - NOT FOUND</h1>
        <h2><Link to='/'>Go Home</Link></h2>
      </div>
      <Footer/>
    </>
    )
}

export default NotFound