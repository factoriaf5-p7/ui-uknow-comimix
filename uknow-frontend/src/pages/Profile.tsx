import React from 'react'
import {ProfileForm} from '../components/forms/ProfileForm'
import Navbar from '../components/navbar/Navbar'
import Footer from '../components/footer/Footer'

function Profile() {
  return (
    <><Navbar />
    <ProfileForm />
    <Footer />
    </>
  )
}

export default Profile