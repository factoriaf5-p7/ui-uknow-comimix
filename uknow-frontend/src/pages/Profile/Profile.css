.MainProfileDiv {
  height: 100vh;
  width: 100vw;
  background: #DEA01E;
  overflow-x: hidden;
}
.MainProfileDiv .profile-container {
  width: 90%;
  height: 100vh;
  background: white;
  margin: auto;
}
.MainProfileDiv .profile-container .top-portion {
  height: 45%;
  width: 100%;
  position: relative;
}
.MainProfileDiv .profile-container .top-portion .user-profile-bg-image {
  height: 100%;
  width: 100%;
  background: #DEA01E;
  overflow: hidden;
}
.MainProfileDiv .profile-container .top-portion .user-profile-bg-image #prf-bg-img {
  height: 100vh;
  width: 100vw;
  margin-top: -12rem;
}
.MainProfileDiv .profile-container .top-portion .user-profile-img {
  position: absolute;
  height: 60%;
  width: 15%;
  border-radius: 50%;
  transform: translateX(50%);
  right: 50%;
  background-color: #124873;
  bottom: -2rem;
  border: 2px solid white;
}
.MainProfileDiv .profile-container .top-portion .user-profile-img #prf-img {
  height: 100%;
  width: 100%;
  -o-object-fit: cover;
     object-fit: cover;
  border-radius: 50%;
}
.MainProfileDiv .profile-container .userNam {
  text-align: center;
  padding-top: 2rem;
  width: 100%;
  position: absolute;
  font-size: 2rem;
  color: black;
}
.MainProfileDiv .profile-container .bottom-portion {
  display: flex;
  justify-content: center;
  flex-direction: row;
  width: 100%;
  margin-top: 5rem;
  text-align: center;
  color: black;
}
.MainProfileDiv .profile-container .bottom-portion .right-side {
  flex: 0.9;
  display: flex;
  justify-content: space-between;
  height: 90%;
  margin-top: 5rem;
  color: black;
}
.MainProfileDiv .profile-container .bottom-portion .left-side {
  flex: 0.9;
  display: flex;
  justify-content: space-between;
  margin-top: 5rem;
  box-shadow: 1rem 1rem 2rem 1rem lightgray;
  margin: 2rem 2rem 2rem 0rem;
  padding: 1rem;
  border-radius: 0.5rem;
}
.MainProfileDiv .image-input {
  position: absolute;
  transform: translateX(150%);
  right: 53%;
}
.MainProfileDiv .form {
  height: 50%;
  margin-top: 1rem;
  width: 100%;
}/*# sourceMappingURL=Profile.css.map */