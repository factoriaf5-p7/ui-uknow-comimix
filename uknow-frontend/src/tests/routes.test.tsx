import { render } from '@testing-library/react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import Hero from '../pages/Hero';
import Home from '../pages/Home';
import Course from '../pages/Course';
import Dashboard from '../pages/Dashboard';
import Profile from '../pages/Profile';
import Login from '../pages/Login';
import Register from '../pages/Register';
import Recover from '../pages/Recover';
import UnderConstruction from '../pages/UnderConstruction';
import NotFound from '../pages/NotFound';

describe('Router renders the correct pages', () => {

test('Snapshot test for Hero page', () => {
    const { asFragment } = render(
      <Router>
        <Routes>
          <Route path="/" element={<Hero />} />
        </Routes>
      </Router>
    );
  
    expect(asFragment()).toMatchSnapshot();
  });

  test('Snapshot test for Home page', () => {
    const { asFragment } = render(
      <Router>
        <Routes>
          <Route path="/home" element={<Home />} />
        </Routes>
      </Router>
    );
  
    expect(asFragment()).toMatchSnapshot();
  });

  test('Snapshot test for Course page', () => {
    const { asFragment } = render(
      <Router>
        <Routes>
          <Route path="/course/:_id" element={<Course />} />
        </Routes>
      </Router>
    );
  
    expect(asFragment()).toMatchSnapshot();
  });

  test('Snapshot test for Dashboard page', () => {
    const { asFragment } = render(
      <Router>
        <Routes>
          <Route path="/dashboard" element={<Dashboard />} />
        </Routes>
      </Router>
    );
  
    expect(asFragment()).toMatchSnapshot();
  });

  test('Snapshot test for Profile page', () => {
    const { asFragment } = render(
      <Router>
        <Routes>
          <Route path="/profile" element={<Profile />} />
        </Routes>
      </Router>
    );
  
    expect(asFragment()).toMatchSnapshot();
  });

  test('Snapshot test for Login page', () => {
    const { asFragment } = render(
      <Router>
        <Routes>
          <Route path="/login" element={<Login />} />
        </Routes>
      </Router>
    );
  
    expect(asFragment()).toMatchSnapshot();
  });
  test('Snapshot test for Register page', () => {
    const { asFragment } = render(
      <Router>
        <Routes>
          <Route path="/signup" element={<Register />} />
        </Routes>
      </Router>
    );
  
    expect(asFragment()).toMatchSnapshot();
  });
  test('Snapshot test for Recover page', () => {
    const { asFragment } = render(
      <Router>
        <Routes>
          <Route path="/recover" element={<Recover />} />
        </Routes>
      </Router>
    );
  
    expect(asFragment()).toMatchSnapshot();
  });
  test('Snapshot test for Under Construction page', () => {
    const { asFragment } = render(
      <Router>
        <Routes>
          <Route path="/under-construction" element={<UnderConstruction />} />
        </Routes>
      </Router>
    );
  
    expect(asFragment()).toMatchSnapshot();
  });
  
  test('Snapshot test for Not Found page', () => {
    const { asFragment } = render(
      <Router>
        <Routes>
          <Route path="*" element={<NotFound />} />
        </Routes>
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});