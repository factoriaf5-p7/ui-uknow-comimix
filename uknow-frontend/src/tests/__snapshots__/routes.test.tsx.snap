// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Router renders the correct pages > Snapshot test for Course page 1`] = `<DocumentFragment />`;

exports[`Router renders the correct pages > Snapshot test for Dashboard page 1`] = `<DocumentFragment />`;

exports[`Router renders the correct pages > Snapshot test for Home page 1`] = `<DocumentFragment />`;

exports[`Router renders the correct pages > Snapshot test for Login page 1`] = `<DocumentFragment />`;

exports[`Router renders the correct pages > Snapshot test for Profile page 1`] = `<DocumentFragment />`;

exports[`Router renders the correct pages > Snapshot test for Recover page 1`] = `<DocumentFragment />`;

exports[`Router renders the correct pages > Snapshot test for Register page 1`] = `<DocumentFragment />`;

exports[`Router renders the correct pages > Snapshot test for Under Construction page 1`] = `<DocumentFragment />`;
